src/gui_common/MicheDetailsPanel.cs

```diff
@@ -0,0 +1,99 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Shows various details a bout a species for the player
+/// </summary>
+public partial class MicheDetailsPanel : MarginContainer
+{
+    [Export]
+    public NodePath? MicheDetailsLabelPath;
+
+    public WorldGenerationSettings WorldSettings = null!;
```

Comment:
This should be a property and if this is set externally I would make this nullable to make the compiler force adding right checks that this class instance is setup correctly.